src/components/Request/Pay/Pay.tsx

```diff
@@ -75,56 +87,59 @@ export const PayRequestLink = () => {
                 setLinkState('ALREADY_PAID')
                 return
             }
+            setLinkState('PAY')
+        } catch (error) {
+            console.error('Failed to fetch request link details:', error)
+        }
+    }
 
-            // Fetch token price
-            const tokenPrice = await utils.fetchTokenPrice(
-                requestLinkDetails.tokenAddress.toLowerCase(),
-                requestLinkDetails.chainId
-            )
-            tokenPrice && setTokenPrice(tokenPrice?.price)
+    useEffect(() => {
```

Comment:
Question: what's the reasoning behind this code? Would  benefit from a comment as to why we need it, I'm not sure I fully get it

Reply:
for what I gathered is to prevent loading on server side, dont know if it still needed with "use client"

Reply:
for what I gathered is to prevent loading on server side, dont know if it still needed with "use client"